File: Java/src/main/java/services/FlightServices.java
package services;

import model.Flight;
import model.validators.FlightValidator;
import repository.FlightRepository;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Collection;

public class FlightServices {
    private FlightRepository flightRepository;
    private FlightValidator flightValidator;

    public FlightServices(FlightRepository flightRepository,FlightValidator flightValidator) {
        this.flightRepository = flightRepository;
        this.flightValidator = flightValidator;
    }

    public void addFlight(String destination, LocalDateTime departure, String airport, Integer availableSeats){
        Flight saveFlight = new Flight(destination,departure,airport,availableSeats);
        flightValidator.validate(saveFlight);
        flightRepository.add(saveFlight);
    }

    public void deleteFlight(String destination, LocalDateTime departure, String airport, Integer availableSeats,Integer Id){
        Flight removeFlight = new Flight(destination,departure,airport,availableSeats);
        removeFlight.setID(Id);
        flightRepository.delete(removeFlight);
    }

    public void updateFlight(String destination, LocalDateTime departure, String airport, Integer availableSeats,Integer Id){
        Flight upFlight = new Flight(Id,destination,departure,airport,availableSeats);
        flightValidator.validate(upFlight);
        flightRepository.update(upFlight, Id);
    }

    public Flight findFlightByID(Integer Id){
        return flightRepository.findByID(Id);
    }

    public Iterable<Flight> findAllFlights(){
        return flightRepository.findAll();
    }

    public Collection<Flight> getAllFlights(){
        return flightRepository.getAll();
    }

    public Collection<Flight> filterFlightsByDestinationAndDeparture(String destination, LocalDate departure){
        Collection<Flight> filterFlights = flightRepository.filterFlightByDestinationAndDeparture(destination, departure);
        return filterFlights;
    }

    public Collection<Flight> filterFlightByAvailableSeats(){
        Collection<Flight> filterFlights = flightRepository.filterFlightByAvailableSeats();
        return filterFlights;
    }
}

File: ProiectCS/travelServer/services/FlightServices.cs
using System;
using System.Collections.Generic;
using travelModel;
using travelModel.validators;
using travelPersistence;

namespace travelServer.services
{
    public class FlightServices
    {
        private FlightRepository flightRepository;
        private Validator<Flight> flightValidator;

        public FlightServices(FlightRepository flightRepository, Validator<Flight> flightValidator)
        {
            this.flightRepository = flightRepository;
            this.flightValidator = flightValidator;
        }

        public void addFlight(string destination, DateTime departure, string airport, int availableSeats)
        {
            Flight saveFlight = new Flight(destination, departure, airport, availableSeats);
            flightValidator.validate(saveFlight);
            flightRepository.Save(saveFlight);
        }

        public void deleteFlight(string destination, DateTime departure, string airport, int availableSeats, int Id)
        {
            Flight removeFlight = new Flight(Id, destination, departure, airport, availableSeats);
            flightRepository.Delete(removeFlight);
        }
        
        public void updateFlight(string destination, DateTime departure, string airport, int availableSeats, int Id)
        {
            Flight upFlight = new Flight(Id, destination, departure, airport, availableSeats);
            flightValidator.validate(upFlight);
            flightRepository.Update(upFlight, Id);
        }

        public Flight findFlightByID(int Id)
        {
            return flightRepository.FindById(Id);
        }

        public IEnumerable<Flight> findALL()
        {
            return flightRepository.FindAll();
        }

        public IEnumerable<Flight> filterFlightsByDestinationAndDeparture(string destination, DateTime departure)
        {
            IEnumerable<Flight> filteredFlights =
                flightRepository.filterFlightByDestinationAndDeparture(destination, departure);
            return filteredFlights;
        }

        public IEnumerable<Flight> filterFlightByAvailableSeats()
        {
            IEnumerable<Flight> filteredFlights = flightRepository.filterFlightByAvailableSeats();
            return filteredFlights;
        }
    }
}